src/main/java/com/github/_1c_syntax/utils/CaseInsensitivePattern.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is a part of 1c-syntax utils.
+ *
+ * Copyright © 2018-2020
+ * Alexey Sosnoviy <labotamy@gmail.com>, Nikita Gryzlov <nixel2007@gmail.com> and contributors
+ *
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ *
+ * 1c-syntax utils is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3.0 of the License, or (at your option) any later version.
+ *
+ * 1c-syntax utils is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with 1c-syntax utils.
+ */
+package com.github._1c_syntax.utils;
+
+
+import lombok.experimental.UtilityClass;
+
+import java.util.regex.Pattern;
+
+@UtilityClass
+public class CaseInsensitivePattern {
+
+  public Pattern compile(String regex) {
```

Comment:
```suggestion
  public static Pattern compile(String regex) {
```

Reply:
оно же уже @UtilityClass?

Reply:
хм... вообще утитилити класс делает класс final и с приватным конструктором. а вот развешивает ли оно модификаторы на все остальные методы - не знаю, надо читать.

Reply:
/**
 * An annotation to create utility classes.
 * 
 * If a class is annotated with {@code @UtilityClass}, the following things happen to it:<ul>
 * <li>It is marked final.</li>
 * <li>If any constructors are declared in it, an error is generated. Otherwise, a private no-args constructor is generated; it throws a {@code UnsupportedOperationException}.</li>
 * <li>All methods, inner classes, and fields in the class are marked static.</li>
 * <li><em>WARNING:</em> Do not use non-star static imports to import these members; javac won't be able to figure it out. Use either:
 *    <code>import static ThisType.*;</code> or don't static-import.</li>
 * </ul>
 */

Reply:
Оу. Тогда это объясняет, почему работало в бсл лс. Спасибо!